Single labelled curried argument in createElement

I tried creating a new component like this, with a single labelled curried argument (similar to what's in the [documentation](https://github.com/reasonml/reason-react/blob/3abf5a04f87a691c5b5ad9b1a8c3b985b835bdfa/documentation.md), but with only one argument):

```ocaml
/* test.re */
module Test = {
  include ReactRe.Component;

  let name = "Test";
  type props = {
    title: string
  };

  let render {props} => {
    <div>
      <h1>(ReactRe.stringToElement props.title)</h1>
    </div>
  };
};

include ReactRe.CreateComponent Test;

let createElement ::title => wrapProps {title};

/* app.re */

/* ... */
  <Test
    title="a title"
  />;
/* ... */
```

This fails with the following error:

```sh
Event change
>>>> Start compiling
Rebuilding since [ 'change : test.re' ]
BSB check build spec : OK 
ninja.exe -C lib/bs 
ninja: Entering directory `lib/bs'
[2/2] Building src/todomvc/test.mlast.d
[2/2] Building src/todomvc/app.cmj /Users/karlhorky/projects/reason-react-experiments/lib/js/src/todomvc/app.js src/todomvc/app.cmi
FAILED: src/todomvc/app.cmj /Users/karlhorky/projects/reason-react-experiments/lib/js/src/todomvc/app.js src/todomvc/app.cmi 
/Users/karlhorky/projects/reason-react-experiments/node_modules/bs-platform/bin/bsc.exe -bs-package-name reason-react-example  -bs-package-output commonjs:lib/js/src/todomvc -bs-assume-no-mli -bs-no-builtin-ppx-ml -bs-no-implicit-include -I /Users/karlhorky/projects/reason-react-experiments/node_modules/reason-react/lib/ocaml -I /Users/karlhorky/projects/reason-react-experiments/node_modules/reason-js/lib/ocaml -I src/interop -I src/logo -I src/simple -I src/todomvc -I src  -nostdlib -I /Users/karlhorky/projects/reason-react-experiments/node_modules/bs-platform/lib/ocaml -no-alias-deps -color always -w -40+6+7+27+32..39+44+45 -o src/todomvc/app.mlast -c  src/todomvc/app.mlast 
File "/Users/karlhorky/projects/reason-react-experiments/src/todomvc/app.re", line 144, characters 19-31:
Error: This expression has type string but an expression was expected of type
         Test.Test.props
ninja: build stopped: subcommand failed.
>>>> Finish compiling
```

---

It does, however, work with two labelled curried arguments:

```ocaml
/* test.re */
module Test = {
  include ReactRe.Component;

  let name = "Test";
  type props = {
    title: string,
    description: string
  };

  let render {props} => {
    <div>
      <h1>(ReactRe.stringToElement props.title)</h1>
      (ReactRe.stringToElement props.description)
    </div>
  };
};

include ReactRe.CreateComponent Test;

let createElement ::title ::description => wrapProps {title, description};

/* app.re */

/* ... */
  <Test
    title="a title"
    description="a description"
  />;
/* ... */
```

--- a/documentation.md
+++ b/documentation.md
@@ -120,10 +120,7 @@ module MyComponent = {
   /*...*/
 };
 
-/* This is the equivalent of `React.createClass` */
 include ReactRe.CreateComponent MyComponent;
-
-/* The actual call exposed to consumers, via JSX */
 let createElement ::foo => wrapProps {foo: foo};
 ```
 

--- a/documentation.md
+++ b/documentation.md
@@ -90,7 +90,7 @@ The component API uses OCaml's module system rather than our own `createClass`-l
 See the `examples/` folder. The components declaration structure should look quite familiar to those who've worked with ReactJS. To declare a React component class, you'd create a normal OCaml module and include some pre-declared module definitions. Here's the basic version:
 
 ```reason
-/* This is like the object declaration part of RaectJS' `React.createClass(theObjectSpec)` */
+/* This is like the object declaration part of ReactJS' `React.createClass(theObjectSpec)` */
 module MyComponent = {
   include ReactRe.Component.Stateful;
   let name = "MyComponent";
@@ -106,6 +106,25 @@ include ReactRe.CreateComponent MyComponent;
 
 /* The actual call exposed to consumers, via JSX */
 let createElement ::foo ::bar=? => wrapProps {foo, bar};
+```
+
+Or, with a single prop:
+
+```reason
+module MyComponent = {
+  include ReactRe.Component.Stateful;
+  let name = "MyComponent";
+  type props = {
+    foo: string,
+  };
+  /*...*/
+};
+
+/* This is the equivalent of `React.createClass` */
+include ReactRe.CreateComponent MyComponent;
+
+/* The actual call exposed to consumers, via JSX */
+let createElement ::foo => wrapProps {foo: foo};
 ```
 
 #### Last Two Expressions

--- a/documentation.md
+++ b/documentation.md
@@ -122,7 +122,7 @@ module MyComponent = {
 
 include ReactRe.CreateComponent MyComponent;
 let createElement ::foo => wrapProps {foo: foo};
-// ^ Note: there's no prop punning for records with a single field. Careful! `{foo}` has a different meaning.
+// ^ Note: there's no record field punning sugar for records with a single field. Careful! `{foo}` has a different meaning.
 ```
 
 #### Last Two Expressions

--- a/documentation.md
+++ b/documentation.md
@@ -114,9 +114,7 @@ Or, with a single prop:
 module MyComponent = {
   include ReactRe.Component.Stateful;
   let name = "MyComponent";
-  type props = {
-    foo: string,
-  };
+  type props = {foo: string};
   /*...*/
 };
 

--- a/documentation.md
+++ b/documentation.md
@@ -121,7 +121,7 @@ module MyComponent = {
 };
 
 include ReactRe.CreateComponent MyComponent;
-let createElement ::foo => wrapProps {foo: foo};
+let createElement ::foo => wrapProps {foo: foo}; // This syntax is necessary for wrapping a single prop
 ```
 
 #### Last Two Expressions

--- a/documentation.md
+++ b/documentation.md
@@ -121,7 +121,8 @@ module MyComponent = {
 };
 
 include ReactRe.CreateComponent MyComponent;
-let createElement ::foo => wrapProps {foo: foo}; // This syntax is necessary for wrapping a single prop
+let createElement ::foo => wrapProps {foo: foo};
+// ^ Note: there's no prop punning for records with a single field. Careful! `{foo}` has a different meaning.
 ```
 
 #### Last Two Expressions

